pkg/caching/caching.go

```diff
@@ -39,17 +46,55 @@ func (i InMemoryHttpTtlCache) Delete(key string) {
 var _ storage.HttpTtlCache = (*InMemoryHttpTtlCache)(nil)
 
 func NewInMemoryHttpTtlCache(
-	memoryLimit string,
+	memoryLimitBytes int64,
 	maxAge time.Duration,
 ) *InMemoryHttpTtlCache {
-	q, err := resource.ParseQuantity(memoryLimit)
-	if err != nil {
-		panic(err)
+	ttlCache := ccache.New(ccache.Configure().MaxSize(memoryLimitBytes).ItemsToPrune(15))
+	return &InMemoryHttpTtlCache{
+		cache:  ttlCache,
+		maxAge: maxAge,
 	}
-	memoryLimitInt := q.Value()
+}
 
-	ttlCache := ccache.New(ccache.Configure().MaxSize(memoryLimitInt).ItemsToPrune(15))
-	return &InMemoryHttpTtlCache{
+type InMemoryGrpcTtlCache struct {
+	cache *ccache.Cache
+
+	maxAge time.Duration
+	// expired is a channel that is used to notify the cache
+	// that an item has to be delete
+}
+
+func (i InMemoryGrpcTtlCache) MaxAge() time.Duration {
+	return i.maxAge
+}
+
+func (i InMemoryGrpcTtlCache) Get(key string) (req proto.Message, ok bool) {
```

Comment:
These args named `req` should probably be named `resp` instead (also below in Set)